src/OpenTelemetry.Instrumentation.StackExchangeRedis/OpenTelemetry.Instrumentation.StackExchangeRedis.csproj

```diff
@@ -12,6 +12,7 @@
 
   <ItemGroup>
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Api\OpenTelemetry.Api.csproj" />
+    <PackageReference Include="fasterflect" Version="3.0.0" />
```

Comment:
@ejsmith We have a few ways we do reflection already in the repo...

* PropertyFetcher: https://github.com/open-telemetry/opentelemetry-dotnet/blob/bd69bd6d29764434bbe08cede8d8003e2f2dccb8/src/OpenTelemetry/DiagnosticSourceInstrumentation/PropertyFetcher.cs#L27
* DynamicMethods: https://github.com/open-telemetry/opentelemetry-dotnet/blob/bd69bd6d29764434bbe08cede8d8003e2f2dccb8/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs#L482
* Expressions: https://github.com/open-telemetry/opentelemetry-dotnet/blob/bd69bd6d29764434bbe08cede8d8003e2f2dccb8/src/OpenTelemetry/Internal/ActivityInstrumentationHelper.cs#L28

My vote would be to use one of those mechanisms and not introduce a new dependency. Unless you can show a benchmark that this lib is way faster and we should be using it everywhere instead 😄

Reply:
PropertyFetcher doesn't work for this scenario. Didn't know about the other ones. Before I try to convert this code to try and use one of those, can I get feedback on if this PR has any chance of getting merged?

Reply:
I'm not opposed to having this. Here are my thoughts...

* The main concern I have is if Redis comes along and gives us an API to use (a la https://github.com/StackExchange/StackExchange.Redis/pull/1454) would having this option in our public API be a problem? I don't think so, we could still use the flag to decide whether or not to take the data from the new API, but "SetCommandKey" could become nonsensical. Maybe rename it something more generalized like "SetVerboseDatabaseStatements" just in case something better comes along? This sort of flows into my next point too...

* The Microsoft IDistributedCache implementation uses `ScriptEvaluate` instead of `HashSet`: https://github.com/dotnet/aspnetcore/blob/709038b45159227a422807f26d6d41dd7707ea10/src/Caching/StackExchangeRedis/src/RedisCache.cs#L25 So this change won't help those users much. Basically, I don't think CommandAndKey is meaningful for all message types. Maybe we could do another PR that captures the script if it is being used? https://github.com/StackExchange/StackExchange.Redis/blob/51d2946b7f1499605d90b95a00bcb8f776ff0b49/src/StackExchange.Redis/RedisDatabase.cs#L3762

* Random thought: The other piece of data that is really nice is whether or not you found something on a GET. So you have cache hit or miss, basically. We're using Redis extensively as a cache at my company we basically just made a wrapper so we could push the key being retrieved and whether or not it was a hit or a miss (on a retrieval) into our traces. Did you investigate at all whether we could do that reflectively too? Doesn't need to be this PR of course, just curious.

Reply:
Changed the option name to `SetVerboseDatabaseStatements` as requested. Added support for putting the scripts into the DbStatement. Removed dependency on Fasterflect.

I looked into trying to see if I could indicate if the command was a cache hit or miss, but I didn't see any way to do so.

Reply:
Also @CodeBlanch, I don't know what the deal is with the code coverage. I guess maybe the Redis instrumentation tests aren't running?

Reply:
The Redis integration tests don't run during code coverage which is why the coverage went down. There isn't really a way to add better test coverage for this. I tried to get the Redis tests to run during the code coverage check, but Redis can't run in Windows.

Reply:
>  Changed the option name to SetVerboseDatabaseStatements as requested. Added support for putting the scripts into the DbStatement. Removed dependency on Fasterflect.

Awesome thanks!

> I looked into trying to see if I could indicate if the command was a cache hit or miss, but I didn't see any way to do so.

Might not be possible, not sure. We can tackle in another PR.

> I don't know what the deal is with the code coverage.

The code coverage check isn't required to pass I don't think we need to worry about it. /cc @cijothomas